test_suite/generic/test_generic.py

```diff
@@ -151,6 +152,27 @@ def test_inittab_and_systemd(self, host):
                     assert 'si::sysinit:/etc/rc.d/rc.sysinit' in host.check_output("grep '^si:' /etc/inittab"), \
                         'Unexpected default inittab "id"'
 
+    def test_release_version(self, host, instance_data):
+        '''
+        Check if rhel provider matches /etc/redhat-release and ami name
+        '''
+        if test_lib.is_rhel_atomic_host(host):
+            pytest.skip('Not run in atomic images')
+
+        cloud_image_name = instance_data['name']
+        product_version = host.system_info.release
+
+        release_file = 'redhat-release'
+        if host.system_info.distribution == 'fedora':
+            release_file = 'fedora-release'
+
+        with host.sudo():
+            package_release_version = host.check_output("sudo rpm -q --qf '%{VERSION}' --whatprovides " + release_file)
```

Comment:
No need to call sudo with sudo.

```suggestion
            package_release_version = host.check_output("rpm -q --qf '%{VERSION}' --whatprovides " + release_file)
```